samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -110,7 +113,18 @@ private enum SupportedFileType
                     return this.BadRequest($"Unsupported file type: {fileType}");
             }
 
-            await this.ParseDocumentContentToMemoryAsync(kernel, fileContent, documentImportForm);
+            var existingMemorySource = (await this._chatMemorySourceRepository.FindByNameAsync(formFile.FileName)).FirstOrDefault();
```

Comment:
Is the name the unique identifier for each source?

imo I think we should treat each uploaded document as its own artifact, especially since we're saving the content of each document to memory and the bot can still reference information that may have been lost between 2 versions in its response. 

Also thinking about the multiuser scenario @TaoChenOSU's about to light up, two users could upload 2 separate documents with the same name and they could be different files

Reply:
+ we're not offering any kind of version control for these sources so I don't think we should allow overrides

Reply:
yes, the name is currently the unique identifier. there was a to-do item from the handoff with Vivien to support upsert instead of create, and I took the same approach that Teams does in that you can only have one document with the same name. 

I think your point is definitely a valid one tho, especially given the multiuser case and that each document can have different data. I can make the change to support allowing duplicate naming hrere

Reply:
updated to allow duplicates.